api/compaction_manager.cc

```diff
@@ -61,6 +74,31 @@ void set_compaction_manager(http_context& ctx, routes& r) {
         });
     });
 
+    cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
+        return ctx.db.map_reduce0([&ctx](database& db) {
+            std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash> tasks;
+            return do_for_each(db.get_column_families(), [&tasks](const std::pair<utils::UUID, seastar::lw_shared_ptr<table>>& i) {
+                table& cf = *i.second.get();
+                tasks[std::make_pair(cf.schema()->ks_name(), cf.schema()->cf_name())] = cf.get_compaction_strategy().estimated_pending_compactions(cf);
+                return make_ready_future<>();
+            }).then([&tasks] {
+                return tasks;
+            });
+        }, std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), sum_panding_tasks).then(
+                [](const std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& task_map) {
+            std::vector<cm::pending_compaction> res;
+            res.reserve(task_map.size());
+            for (auto i : task_map) {
+                cm::pending_compaction task;
+                task.ks = i.first.first;
+                task.cf = i.first.second;
+                task.task = i.second;
+                res.emplace_back(task);
+            }
+            return make_ready_future<json::json_return_type>(res);
+        });
```

Comment:
std::move()

Reply:
ok

Reply:
where is it?

Reply:
I've added the std::move to task, there's no point in adding it to res, the constructor of json_return_type takes it by reference and translate it to the return string.

Reply:
so say something.